packages/oas/test/index.test.ts

```diff
@@ -1322,6 +1322,36 @@ describe('Oas', () => {
         expect(operation).toBeUndefined();
       });
 
+      it('should be able to find a match on a url with a non-default port', () => {
```

Comment:
Do you mind adding a few more hard test here ? I.E.  a file that has 2 servers and like 3 different vars? One in domain, on for port, and one for path

Reply:
we do have a lot of those kinds of tests in this file already, feel free to take a look in these places and let me know if you can think of any gaps:

https://github.com/readmeio/oas/blob/f2d3b32595993807e4f65232dd288adec6310871/packages/oas/test/index.test.ts#L867-L945 

https://github.com/readmeio/oas/blob/f2d3b32595993807e4f65232dd288adec6310871/packages/oas/test/index.test.ts#L1242-L1397

i'm truthfully still kinda stumped how the example you provided wasn't working given how thorough our tests already are 🧐

Reply:
ahh perf!